mongo/client.go

```diff
@@ -358,243 +358,21 @@ func (c *Client) endSessions(ctx context.Context) {
 }
 
 func (c *Client) configure(opts *options.ClientOptions) error {
```

Comment:
Optional: Now that the amount of code in `configure` has been significantly reduced, consider merging the remaining code into `NewClient`.